### ISSUE TITLE
etcd 集群链接 报错context canceled

### ISSUE THREAD

**Issue by info193** [etcd-io/etcd#16250]
### Bug report criteria

- [ ] This bug report is not security related, security issues should be disclosed privately via security@etcd.io.
- [X] This is not a support request, support requests should be raised in the etcd [discussion forums](https://github.com/etcd-io/etcd/discussions).
- [X] You have read the etcd [bug reporting guidelines](https://github.com/etcd-io/etcd/blob/main/Documentation/contributor-guide/reporting_bugs.md).
- [X] Existing open issues along with etcd [frequently asked questions](https://etcd.io/docs/latest/faq) have been checked and this is not a duplicate.

### What happened?

etcd集群链接报错，集群链接配置 RejectOldCluster=true检测 checkVersion 有一个链接成功就跳过其它节点链接，直接cancal取消上下文，导致链接报错。
![image](https://github.com/etcd-io/etcd/assets/50605978/b6c31b0f-3873-4ebb-b881-1065e5a9126a)

### What did you expect to happen?

集群链接报错
![image](https://github.com/etcd-io/etcd/assets/50605978/61fbf3d3-9c04-4438-af56-fee1e3cd671e)

### How can we reproduce it (as minimally and precisely as possible)?

在本机环境，关闭etcd集群，然后启动，启动完成，立即启动运行etcd链接代码。注：在第一次运行的时候会报错，后面在重试启动未报错。
注：重复复现流程，关闭etcd集群在启动etcd集群，然后运行etcd链接代码。
##etcd链接代码
![image](https://github.com/etcd-io/etcd/assets/50605978/f6af1b4f-6388-4a95-897e-c65d69fa8c19)

### Etcd version (please run commands below)

<details>

```console
$ etcd --version
etcd Version: 3.5.9
Git SHA: bdbbde9
Go Version: go1.19.10
Go OS/Arch: linux/amd64

$ etcdctl version
etcdctl version: 3.5.9
API version: 3.5
```

</details>

### Etcd configuration (command line flags or environment variables)

<details>

# paste your configuration here

</details>

### Etcd debug information (please run commands below, feel free to obfuscate the IP address or FQDN in the output)

<details>

```console
$ etcdctl member list -w table
# paste output here

$ etcdctl --endpoints=<member list> endpoint status -w table
# paste output here
```

</details>

### Relevant log output

```Shell
{"level":"warn","ts":"2023-07-15T11:32:45.366+0800","logger":"etcd-client","caller":"v3@v3.5.5/retry_interceptor.go:62","msg":"retry
ing of unary invoker failed","target":"etcd-endpoints://0xc000203340/192.168.7.75:2379","attempt":0,"error":"rpc error: code = Cance
led desc = context canceled"}
{"level":"warn","ts":"2023-07-15T11:32:45.366+0800","logger":"etcd-client","caller":"v3@v3.5.5/retry_interceptor.go:62","msg":"retrying of unary invoker failed","target":"etcd-endpoints://0xc000203340/192.168.7.75:2379","attempt":0,"error":"rpc error
ing of unary invoker failed","target":"etcd-endpoints://0xc000203340/192.168.7.75:2379","attempt":0,"error":"rpc error: code = Cance
led desc = context canceled"}
```

**Comment by info193** [comment]
文件路径 go.etcd.io\etcd\client\v3@v3.5.5\client.go    512行

**Comment by CaojiamingAlan** [comment]
Seems that #14368 fixs this issue, but it is not backported to v3.5. 

@info193 Thank you for raising this issue. Next time **please please** try to :
**1. Use English** 
**2. Use copied code rather than snapshot**
**3. Use [permalink](https://github.com/etcd-io/etcd/blob/bdbbde998b7ed434b23676530d10dbd601c4a7c0/client/v3/client.go#L512) instead of using path**

**Comment by fuweid** [comment]
@info193 would you mind to test it with latest release-3.5 branch? if the backport works, I will close it. Thanks

**Comment by serathius** [comment]
Should this be backported to v3.4?

**Comment by jmhbnz** [comment]
Closed - 3.4 backport was completed in https://github.com/etcd-io/etcd/pull/16849.
